src/main/java/duke/command/MarkCommand.java

```diff
@@ -0,0 +1,59 @@
+package duke.command;
+
+import duke.logic.DukeException;
+import duke.logic.Storage;
+import duke.task.Task;
+import duke.logic.TaskList;
+import duke.logic.Ui;
+
+/**
+ * Represents a command that marks or unmarks a task.
+ *
+ * @author Peter
+ */
+public class MarkCommand extends Command {
+    /**
+     * Index of task in a list of tasks that is to be marked.
+     */
+    private final int index;
+
+    /**
+     * Boolean flag that determines whether to mark or unmark task.
+     */
+    private final boolean isMark;
```

Comment:
```suggestion
    private final boolean isMarked;
```
Perhaps the above suggestion would be good here?